From da6368e7a5eeeb9aaac7233ac392a6b12f56d434 Mon Sep 17 00:00:00 2001
From: Tomasz Figa <t.figa@samsung.com>
Date: Thu, 28 Feb 2013 19:42:59 +0100
Subject: [PATCH 0147/1302] ARM: dts: exynos4210-trats: Add nodes for audio
 hardware

This patch adds nodes for audio hardware present on Trats board.

Signed-off-by: Tomasz Figa <t.figa@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 arch/arm/boot/dts/exynos4210-trats.dts | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/arch/arm/boot/dts/exynos4210-trats.dts b/arch/arm/boot/dts/exynos4210-trats.dts
index d87f19a..7061c88 100644
--- a/arch/arm/boot/dts/exynos4210-trats.dts
+++ b/arch/arm/boot/dts/exynos4210-trats.dts
@@ -448,6 +448,22 @@
 		};
 	};
 
+	i2c@138C0000 {
+		samsung,i2c-sda-delay = <100>;
+		samsung,i2c-slave-addr = <0x10>;
+		samsung,i2c-max-bus-freq = <400000>;
+		pinctrl-0 = <&i2c6_bus>;
+		pinctrl-names = "default";
+		status = "okay";
+
+		mc1n2: codec@3a {
+			compatible = "yamaha,mc1n2";
+			reg = <0x3a>;
+			vadc-supply = <&vadc_reg>;
+			gpios = <&gpe1 4 0>, <&gpe2 0 0>;
+		};
+	};
+
 	usbphy@125B0000 {
 		status = "okay";
 	};
@@ -594,6 +610,20 @@
 		};
 	};
 
+	i2s0: i2s@03830000 {
+		pinctrl-0 = <&i2s0_bus>;
+		pinctrl-names = "default";
+		status = "okay";
+	};
+
+	sound {
+		compatible = "samsung,u1-platform-snd";
+		clocks = <&clock 2>, <&clock 395>, <&clock 21>;
+		clock-names = "parent", "out-mux", "out";
+		samsung,i2s-controller = <&i2s0>;
+		samsung,audio-codec = <&mc1n2>;
+	};
+
 	cpufreq {
 		/* No freq_table property to use default table. */
 		vdd_arm-supply = <&varm_breg>;
-- 
1.8.3.2

